include/kodlab_mjbots_sdk/joint_base.h

```diff
@@ -0,0 +1,173 @@
+/**
+ * @file jointBase.h
+ * @author Kodlab - J. Diego Caporale (jdcap@seas.upenn.edu)
+ * @brief Abstract class for joints. Agnostic to servo/motor used.
+ * @version 0.1
+ * @date 2022-02-17
+ * 
+ * @copyright BSD 3-Clause License, Copyright (c) 2021 The Trustees of the University of Pennsylvania. All Rights Reserved
+ * 
+ */
+
+#pragma once
+#include <limits>
+
+/**
+ * @brief 
+ * Abstract Base class for any joint used in the robot class.
+ */
+class JointBase {
+
+    protected:
+        // Joint Params
+        float gear_ratio_  = 1.0;   /// external joint gear ratio
+        float zero_offset_ = 0;     /// zero offset [rad]
+        int   direction_   = 1;     /// direction [-1 or 1]
+
+        // Joint State/Commands
+        float position_;    /// position of the joint [rad]
+        float velocity_;    /// velocity of the joint [rad/s]
+        float torque_ = 0;  /// Constrained torque for the joint [N m]
+
+        // Servo State/Commands
+        float servo_position_;      /// position of the servo [rad]
```

Comment:
you should add a comment here explaining the difference between the servo and the joint